### PR TITLE
Keep kafka-ch-dispatcher replicas set by controller

### PR DESCRIPTION
**Pull request by maschmid**

### CODE REVIEW

File: knative-operator/pkg/controller/knativekafka/replicastransform.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+			}
+
+			current := &appsv1.Deployment{}
+			if err := scheme.Scheme.Convert(currentU, current, nil); err != nil {
+				return err
+			}
+
+			// Keep the existing number of replicas in the cluster for the deployment
+			apply.Spec.Replicas = current.Spec.Replicas
+
```
Comment (matzew) [inline knative-operator/pkg/controller/knativekafka/replicastransform.go:36]: Do we need to take care of the `ENV_VARS` too ? 

See:
* https://github.com/knative/operator/issues/393
* https://github.com/knative/operator/pull/394

/cc @lionelvillard

Comment (maschmid) [reply knative-operator/pkg/controller/knativekafka/replicastransform.go:36]: I don't think so, looking at https://github.com/knative-sandbox/eventing-kafka/blob/release-0.20/pkg/channel/consolidated/reconciler/controller/kafkachannel.go#L314  , it should accept the Deployment as is if created by the Operator, as long as the container image matches.

**Comment by maschmid** [comment]
I don't think the TestKafkaBinding has to do anything with the PR, possibly a flake...

**Comment by matzew** [comment]
> I don't think the TestKafkaBinding has to do anything with the PR, possibly a flake...

Nor do I think that ... and I saw it on the "retry...." PR too ... (which bumps to a different version...)

**Comment by maschmid** [comment]
> > I don't think the TestKafkaBinding has to do anything with the PR, possibly a flake...
> 
> Nor do I think that ... and I saw it on the "retry...." PR too ... (which bumps to a different version...)

It's not a "Required" job anyway, so it doesn't block this PR.

**Comment by matzew** [comment]
FWIW.... this PR did not flake fail:

https://github.com/openshift-knative/serverless-operator/pull/855

**Comment by matzew** [comment]
I might cherry pick yours to my other drama here: # 854

**Comment by maschmid** [comment]
> I might cherry pick yours to my other drama here: # 854

sure, go ahead!

**Comment by matzew** [comment]
/close

done via #854   (:cherries: picked)

**Pull request closed without merging**